analyses/malawi/notebooks/read_in_data.py

```diff
@@ -0,0 +1,184 @@
+sys.path.append(path_mod)
+
+from src.indicators.drought import ecmwf_seasonal
+from src.malawi import get_ecmwf_seasonal_data as gesd
+from src.indicators.drought.config import Config
+
+#TODO: not sure which functions to include here and which in the ecmwf_seasonal file
+
+def get_ecmwf_forecast(version: int = 5):
+    """
+    Retrieve the processed dataset with the forecast for each publication date and corresponding lead times
+    Args:
+        version: version of forecast model that was used (only changes once every couple of years)
+    """
+    ecmwf_forecast = ecmwf_seasonal.EcmwfSeasonalForecast()
+    ds_ecmwf_forecast = ecmwf_forecast.read_processed_dataset(
+            country_name=gesd.COUNTRY_NAME,
+            country_iso3=gesd.COUNTRY_ISO3,
+            version=version,
+        )
+    ds_ecmwf_forecast = convert_tprate_precipitation(ds_ecmwf_forecast)
+
+    return ds_ecmwf_forecast
+
+def get_ecmwf_forecast_by_leadtime(version: int = 5):
+    """
+    Reshape dataset to have the time variable as the month during the forecast was valid
+    instead of the month the forecast was published
+    Args:
+        version: version of forecast model that was used (only changes once every couple of years)
+
+    Returns:
+        dataset with valid month per publication data-leadtime
+    """
+    ds_ecmwf_forecast = get_ecmwf_forecast(version)
+    ds_ecmwf_forecast_dict = dates_per_leadtime(ds_ecmwf_forecast)
+    return convert_dict_to_da(ds_ecmwf_forecast_dict)
+
+#TODO: not sure if this is the best structure, should it instead be inside a class?
+def compute_stats_per_admin(country,adm_level=1):
+    config = Config()
+    parameters = config.parameters(country)
+
+    country_data_raw_dir = os.path.join(config.DATA_DIR, config.RAW_DIR, country)
+    country_data_processed_dir = os.path.join(config.DATA_DIR, config.PROCESSED_DIR, country)
+    adm_boundaries_path = os.path.join(country_data_raw_dir, config.SHAPEFILE_DIR, parameters[f"path_admin{adm_level}_shp"])
+
+    ds = get_ecmwf_forecast_by_leadtime()
+    df=compute_zonal_stats(ds,ds.rio.transform(),adm_boundaries_path,parameters[f"shp_adm{adm_level}c"])
+
+    df.to_csv(os.path.join(country_data_processed_dir,"ecmwf_seasonal_data",f"{parameters['iso3_code']}_seasonal-monthly-single-levels_v5_adm{adm_level}_stats.csv"))
+
+#TODO: create function to retrieve the stats file
+
+def compute_zonal_stats(ds, raster_transform, adm_path,adm_col):
+    # compute statistics on level in adm_path for all dates in ds
+    df_list = []
+    for date in ds.time.values:
+        for leadtime in ds.leadtime.values:
+            for number in ds.number.values:
+                df = gpd.read_file(adm_path)[[adm_col,"geometry"]]
+                ds_date = ds.sel(time=date,number=number,leadtime=leadtime)
+
+                df[["mean_cell", "max_cell", "min_cell"]] = pd.DataFrame(
+                    zonal_stats(vectors=df, raster=ds_date.values, affine=raster_transform, nodata=np.nan))[
```

Comment:
If speed is an issue, I found it can be faster to use rasterio mask, see this issue [here](https://github.com/OCHA-DAP/pa-COVID-model-parameterization/issues/35)

Reply:
Wauw this is great if that speeds up things! Since it is an xarray dataset instead of rasterio it needs some changes, but I think it should be possible. Not the time to look into it now, but will make an issue for it! (also in other scripts we are using `zonal_stats` quite a bit so definitely worth looking into)